### PR TITLE
Separate FrameBuilder into it's own file

### PR DESCRIPTION
**Pull request by mrobinson**
This is the first step toward splitting up tiling.rs into logical
units and will help to reduce merge conflicts in the future.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/834)

### CODE REVIEW

**Comment by mrobinson** [comment]
@glennw or @kvark r?

**Comment by kvark** [comment]
I think the PR closes #587

**Comment by mrobinson** [comment]
@kvark It does help, but I still think there is more work to do. tiling.rs still contains many things that can be moved out. :smile:

File: webrender/src/tiling.rs [hunk 1]
```diff
@@ -860,10 +845,10 @@ impl ClipBatcher {
     }
 }
 
-struct RenderTargetContext<'a> {
-    stacking_context_store: &'a [StackingContext],
-    prim_store: &'a PrimitiveStore,
-    resource_cache: &'a ResourceCache,
+pub struct RenderTargetContext<'a> {
```
Comment (kvark) [inline webrender/src/tiling.rs]: Can we have the fields hidden and only exposed to the module that uses them?
I imagine that both `tiling` and `frame_builder` need to create a `RenderTargetContext`, but do they both need to access the fields?

File: webrender/src/tiling.rs [hunk 2]
```diff
@@ -1693,32 +1670,32 @@ pub struct StackingContextIndex(pub usize);
...
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, Ord, PartialOrd)]
 pub struct ScrollLayerIndex(pub usize);
 
 pub struct ScrollLayer {
-    scroll_layer_id: ScrollLayerId,
-    clip_source: ClipSource,
-    clip_cache_info: Option<MaskCacheInfo>,
-    packed_layer_index: PackedLayerIndex,
-    xf_rect: Option<TransformedRect>,
+    pub scroll_layer_id: ScrollLayerId,
```
Comment (kvark) [inline webrender/src/tiling.rs]: ok, so there is a lot of stuff that is now public
I suppose we can address the privacy concerns in follow-ups

Comment (mrobinson) [inline webrender/src/tiling.rs]: I agree. I think once everything has been moved it makes sense to go back and reduce the visibility of everything by adding methods to the structs.

**Comment by mrobinson** [comment]
Thanks for the reviews.

**Pull request merged**